packages/did/src/DidBatcher/FullDidBuilder.utils.ts

```diff
@@ -13,27 +13,29 @@ import { ApiPromise } from '@polkadot/api'
 import type { DidKey, NewDidKey } from '@kiltprotocol/types'
 import { Crypto, SDKErrors } from '@kiltprotocol/utils'
 
-import { ChainDidPublicKey, formatPublicKey } from '../Did.chain.js'
+import { ChainDidPublicKey, encodePublicKey } from '../Did.chain.js'
 import { isEncryptionKey, isVerificationKey } from '../Did.utils.js'
 
 function computeChainKeyId(publicKey: ChainDidPublicKey): DidKey['id'] {
   return Crypto.hashStr(publicKey.toU8a())
 }
 
 function encodeToChainKey(api: ApiPromise, key: NewDidKey): ChainDidPublicKey {
-  let keyClass: string
   if (isVerificationKey(key)) {
-    keyClass = 'PublicVerificationKey'
-  } else if (isEncryptionKey(key)) {
-    keyClass = 'PublicEncryptionKey'
-  } else {
-    throw new SDKErrors.DidBuilderError('Unsupported key type')
+    return new (api.registry.getOrThrow<ChainDidPublicKey>(
```

Comment:
This duplication is not as minimal as the other one, but I still consider it a reasonable cost

Reply:
yes, I think readability is improved